### PR TITLE
add devcontainer for gh workspaces

### PR DESCRIPTION
**Pull request by aruniverse**
Add a devcontainer so we can properly open the repo in GitHub workspaces; currently if you open the repo in workspaces (maybe when you have a quick small patch or just wanted to review a pr) there is no `rush` so you'd need to clone the repo locally to use, but with this, should be able to develop locally.

### CODE REVIEW

**Review by kckst8 — COMMENTED** [review]
Looks good but where did the need for code spaces come from (no description in the pr)? Is this extension related or just a nice to have addition?

**Comment by aruniverse** [comment]
> Looks good but where did the need for code spaces come from (no description in the pr)? Is this extension related or just a nice to have addition?

Sorry, didnt submit the description last night, added now.

This is just a nice to have, so we can open and use the repo in workspaces (because it'll now include rush) so you don't have to clone/run locally.

**Review by kckst8 — APPROVED** [review]
Works great on Mac. I added a subsection for Windows in the README (and I made the change, committed, and pushed in the container 👍). Feel free to update as needed.

File: .devcontainer/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+### All OS
+
+Update VS Code to at least 1.35.0 (May 2019).
+
+Install/update VS Code extension "Remote - Containers".
+
+## How to use the container
+
+Open your iTwin.js repo in VS Code, click the green area in the status bar, and select VS Code command 'Remote-Containers: Reopen Folder in Container'. Once connected, the green section of the status bar should say something like "Dev Container: imodeljs-linux".
```
Comment (calebmshafer) [inline .devcontainer/README.md:30]: `iTwin.js repo` -> `Viewer repo`

**Review by ben-polinsky — APPROVED** [review]

**Pull request merged**